tests fail to build: error: use of undeclared identifier 'CTL_KERN'

```
/usr/bin/c++ -DHAVE_NLOHMANN_JSON -DJSON_USE_IMPLICIT_CONVERSIONS=1 -Iinclude -isystem /usr/local/include -O2 -pipe -fno-omit-frame-pointer -fstack-protector-strong -fno-strict-aliasing -fno-omit-frame-pointer -O2 -pipe -fno-omit-frame-pointer -fstack-protector-strong -fno-strict-aliasing -fno-omit-frame-pointer -Wunused-parameter -Wextra -Wreorder -Wconversion -Wsign-conversion -march=native -pthread -MD -MT test/CMakeFiles/test_xsystem.dir/test_xsystem.cpp.o -MF test/CMakeFiles/test_xsystem.dir/test_xsystem.cpp.o.d -o test/CMakeFiles/test_xsystem.dir/test_xsystem.cpp.o -c test/test_xsystem.cpp
In file included from test/test_xsystem.cpp:10:
include/xtl/xsystem.hpp:79:23: error: use of undeclared identifier 'CTL_KERN'
        int mib[4] = {CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1};
                      ^
include/xtl/xsystem.hpp:79:33: error: use of undeclared identifier 'KERN_PROC'
        int mib[4] = {CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1};
                                ^
include/xtl/xsystem.hpp:79:44: error: use of undeclared identifier 'KERN_PROC_PATHNAME'
        int mib[4] = {CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1};
                                           ^
include/xtl/xsystem.hpp:80:13: error: use of undeclared identifier 'sysctl'
        if (sysctl(mib, 4, buffer, sizeof(buffer), NULL, 0) != -1)
            ^
4 errors generated.
```

Version: 0.6.23
OS: FreeBSD 12.2

Now it fails like this:

```
In file included from test/test_xsystem.cpp:10:
include/xtl/xsystem.hpp:84:13: error: no matching function for call to 'sysctl'
        if (sysctl(mib, 4, buffer, sizeof(buffer), NULL, 0) != -1)
            ^~~~~~
/usr/include/sys/sysctl.h:1116:5: note: candidate function not viable: no known conversion from 'unsigned long' to 'size_t *' (aka 'unsigned long *') for 4th argument
int     sysctl(const int *, u_int, void *, size_t *, const void *, size_t);
        ^
1 error generated.
```

--- a/include/xtl/xsystem.hpp
+++ b/include/xtl/xsystem.hpp
@@ -29,6 +29,10 @@
 #if defined(__sun)
 #  include <stdlib.h>
 #endif
+#ifdef __FreeBSD__
+#  include <sys/types.h>
+#  include <sys/sysctl.h>
+#endif
 
 #include <cstring>
 #include <string>

